### ISSUE TITLE
Integrate ProtoFlux nested nodes into FrooxEngine

### ISSUE THREAD

**Issue by Frooxius** [Yellow-Dog-Man/Resonite-Issues#564]
### Is your feature request related to a problem? Please describe.

ProtoFlux VM has internal support for nested nodes, which allows defining shared logic and reusing it across many other nodes. However this is not currently exposed to FrooxEngine.

### Describe the solution you'd like

Ability to create and then use nested nodes will be exposed to ProtoFlux visual tools in FrooxEngine. Users will be able to create a node definition (input values, input impulses, output values, output impulses, global refs) and then wire up internal behavior.

This node definition can then be used as part of other node setups, including other nested nodes - or even the nested node itself for recursion.

### Describe alternatives you've considered

N/A

### Additional Context

There will be additional features on top of this later - interfaces, dynamic binding / dependency injection, packaging, versioning and so on. 

This is specifically to get the initial building block in first.

**Comment by ultrawidegamer** [comment]
@Frooxius is this basically the same issue as #58?

**Comment by Frooxius** [comment]
Yes, this would be addressed by this. I'll close the other one. Thanks!

**Comment by JackTheFoxOtter** [comment]
Some components expose reference fields for functions / delegates, like the CustomCanGrabCheck on the Grabbable Component. With nested nodes, will we be able to define ProtoFlux functions that can be used with these?

**Comment by ColinTimBarndt** [comment]
I'd like to add a consideration for the implementation that's been on my mind. Complex creations like Moduprint or games like Lagrange Point or Cloudscape Harvest make use of centralized managers which provide function/method implementations to be called by dynamic impulses. The advantage is that managers can be updated individually such that all other logic then uses the updated methods.

To me, it would be important that references to nested nodes could easily be updated in bulk if the signatures (input/output types) match.

**Comment by JackTheFoxOtter** [comment]
Something I've just had to think about in regards to nested nodes. Some ProtoFlux nodes have an internal state on the data model (like FieldHook, which stores which field is being driven, or DataModelBooleanToggle which stores its value on the data model). Would those nodes be compatible with nested nodes? As a more concrete example, could I create a nested node that internally utilizes the FieldHook node as a reusable piece and use multiple instances of that nested node in different places without them affecting one another?
